Scripts/datatypes/tour.py

```diff
@@ -179,6 +181,24 @@ def choose_secondary_destination(self, cumulative_probs):
                     + numpy.searchsorted(cumulative_probs, self._sec_dest_draw))
         self.position = (self.position[0], self.position[1], dest_idx)
         self.purpose.sec_dest_purpose.attracted_tours[self.mode][dest_idx] += 1
+    
+    def calc_cost(self):
+        """Construct cost and time components from tour dest choice. """
+        impedance = self.purpose.model.impedance[self.mode]
+        time = self._get_cost(impedance["time"])
+        self.cost = self._get_cost(impedance["cost"])
```

Comment:
Will this not cause a `KeyError` for bike and walk?

Reply:
I think this is related to above. Bike and walk return float zeros.